File: app/src/main/java/com/manoj/dlt/utils/FirebaseChildAddedListener.java
package com.manoj.dlt.utils;

import com.google.firebase.database.ChildEventListener;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;

public abstract class FirebaseChildAddedListener implements ChildEventListener
{
    @Override
    public void onChildChanged(DataSnapshot dataSnapshot, String s) { }

    @Override
    public void onChildRemoved(DataSnapshot dataSnapshot) { }

    @Override
    public void onChildMoved(DataSnapshot dataSnapshot, String s) { }

    @Override
    public void onCancelled(DatabaseError databaseError) { }
}

File: app/src/main/kotlin/com/manoj/dlt/utils/FirebaseChildAddedListener.kt
package com.manoj.dlt.utils

import com.google.firebase.database.ChildEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError

abstract class FirebaseChildAddedListener : ChildEventListener {
    override fun onChildChanged(dataSnapshot: DataSnapshot, s: String) {
    }

    override fun onChildRemoved(dataSnapshot: DataSnapshot) {
    }

    override fun onChildMoved(dataSnapshot: DataSnapshot, s: String) {
    }

    override fun onCancelled(databaseError: DatabaseError) {
    }
}
